connector-drawer/src/main/java/com/redhat/cloud/notifications/connector/drawer/DrawerRouteBuilder.java

```diff
@@ -42,17 +44,23 @@ public void configureRoutes() {
                 .to(drawerConnectorConfig.getRecipientsResolverServiceURL() + "/internal/recipients-resolver")
             .to(RECIPIENTS_RESOLVER_RESPONSE_TIME_METRIC + TIMER_ACTION_STOP)
             .process(recipientsResolverResponseProcessor)
-            .split(simpleF("${exchangeProperty.%s}", ExchangeProperty.RESOLVED_RECIPIENT_LIST))
+            .choice().when(shouldDrawerPayloadGenAndSend())
                 .process(drawerPayloadBuilder)
                 .to(log(getClass().getName()).level("INFO").showHeaders(true).showBody(true))
                 .to(direct(CONNECTOR_TO_DRAWER))
+                .log(INFO, getClass().getName(), "Sent Drawer notification " +
+                    "[orgId=${exchangeProperty." + ORG_ID + "}, historyId=${exchangeProperty." + ID + "}]")
             .end()
-            .log(INFO, getClass().getName(), "Sent Drawer notification " +
-                "[orgId=${exchangeProperty." + ORG_ID + "}, historyId=${exchangeProperty." + ID + "}]")
             .to(direct(SUCCESS));
 
         from(direct(CONNECTOR_TO_DRAWER))
             .routeId(CONNECTOR_TO_DRAWER)
             .to(kafka(drawerConnectorConfig.getOutgoingDrawerTopic()));
     }
+
+
+    private Predicate shouldDrawerPayloadGenAndSend() {
```

Comment:
Nitpick: the same could be simpler 😃 
```suggestion
    private Predicate hasRecipients() {
```